sql/src/main/java/io/crate/auth/user/AccessControl.java

```diff
@@ -22,12 +22,24 @@
 
 package io.crate.auth.user;
 
+import io.crate.analyze.AnalyzedStatement;
 import io.crate.exceptions.MissingPrivilegeException;
 
-public interface ExceptionAuthorizedValidator {
+public interface AccessControl {
 
-    /**
-     * Ensures that the given exception is allowed to be thrown for the given user
-     */
-    void ensureExceptionAuthorized(Throwable t) throws MissingPrivilegeException;
+    AccessControl DISABLED = new AccessControl() {
+        @Override
+        public void ensureMayExecute(User user, String defaultSchema, AnalyzedStatement statement) {
+
+        }
+
+        @Override
+        public void ensureMaySee(User user, Throwable t) throws MissingPrivilegeException {
+
+        }
+    };
+
+    void ensureMayExecute(User user, String defaultSchema, AnalyzedStatement statement);
```

Comment:
not sure about the interfaces here, or whether we should keep binding the user into the `AccessControl` and change the getter in the `UserManager` to `UserManager.getAccessControl(user, defaultSchema)`

Reply:
I'd prefer the later: keep binding the user info to the `AccessControl`. This would differentiate thing better imho (e.g. `isSuperuser` -> `AccessControl.DISABLED` and reduce the requirements of the `user` argument on some other methods as well, e.g. `SqlException.forWireTransmission()`

Reply:
Pushed a fixup, see comment here: https://github.com/crate/crate/pull/8720/commits/faa7603232fbae2788e4c3c05ad2ed48aafb7204#diff-9961520d4cf455cc2bb67ad310818d4eR94 why I decided to use `getAccessControl(sessionContext)` instead of `getAccessControl(user, defaultSchema)`